### PR TITLE
[DO NOT MERGE] Rewrite YAPF scripts

### PR DESCRIPTION
**Pull request by alok**
## What do these changes do?

- Fixes YAPF to use existing `.style.yapf` config file
- Runs in parallel
- Easier to read `yapf.sh` (IMO)

## Related issue number

No.

### CODE REVIEW

**Comment by pcmoritz** [comment]
@ericl @richardliaw Are these linting changes ok with you? They look good to me!

**Comment by alok** [comment]
Personally, I'd tweak the `.style.yapf` file a bit. I set mine up to imitate Elm's style guide in the areas where Python has no clear rules, and I think it's easier to read. I just used the pep8 style in this PR because that was what was specified before. I think it's harder to read this way though since it has odd indentation at times.

**Comment by alok** [comment]
Regardless of the particular style used, we should use this updated script since it shaves about 30s off Travis builds.

File: test/actor_test.py [hunk 1]
```diff
@@ -1087,8 +1087,9 @@ def locations_to_intervals_for_many_tasks():
 
         # Create more actors to fill up all the GPUs.
         more_actors = [
-            Actor1.remote() for _ in range(
-                num_local_schedulers * num_gpus_per_scheduler - 1 - 3)
+            Actor1.remote()
+            for _ in range(num_local_schedulers * num_gpus_per_scheduler - 1 -
```
Comment (richardliaw) [inline test/actor_test.py]: this one looks quite odd..

File: python/ray/monitor.py [hunk 2]
```diff
@@ -507,8 +506,8 @@ def run(self):
         log.debug("{} dead local schedulers, {} plasma managers total, {} "
                   "dead plasma managers".format(
                       len(self.dead_local_schedulers),
-                      (len(self.live_plasma_managers) +
-                       len(self.dead_plasma_managers)),
+                      (len(self.live_plasma_managers) + len(
```
Comment (richardliaw) [inline python/ray/monitor.py]: this one is also a little odd..

File: python/ray/experimental/state.py [hunk 3]
```diff
@@ -750,8 +750,8 @@ def micros_rel(ts):
                         "name":
                         "SubmitTask",
                         "args": {},
-                        "id": (parent_info["worker_id"] +
-                               str(micros(min(parent_times))))
+                        "id": (parent_info["worker_id"] + str(
```
Comment (richardliaw) [inline python/ray/experimental/state.py]: not a big fan of these

File: python/ray/experimental/state.py [hunk 4]
```diff
@@ -825,8 +825,8 @@ def micros_rel(ts):
...
```
Comment (richardliaw) [inline python/ray/experimental/state.py]: not a big fan of these

**Review by richardliaw — COMMENTED** [review]
Some comments

**Comment by alok** [comment]
@richardliaw See the latest commit where I apply the style specified in the config file. I went in and manually tweaked some code by adding trailing commas so the formatter would do the right thing. I think it fixed all your comments and hopefully looks fine for the rest. If not, feel free to revert it.

**Comment by richardliaw** [comment]
OK I think there's too much vertical space as of this most recent commit; would you be willing to revert?

**Comment by alok** [comment]
Done.

File: python/ray/experimental/state.py (hunk 4 above)
Comment (richardliaw) [reply python/ray/experimental/state.py]: @alok can we revert this?

File: python/ray/monitor.py (hunk 2 above)
Comment (richardliaw) [reply python/ray/monitor.py]: @alok can we revert this?

**Review by richardliaw — CHANGES_REQUESTED** [review]
two requests to change; otherwise looks good overall.

**Comment by alok** [comment]
@richardliaw I'd rather tweak the formatting config until it ends up looking pretty to avoid ever having to worry about manual formatting. I want it to be reliable enough to become a git pre-push hook so no one ever has to comment about formatting again.

**Comment by alok** [comment]
@richardliaw can you try out this style? i think it fixes (most) of your issues with the current one.

**Comment by richardliaw** [comment]
Hey @alok how did this become a 4k line PR?

**Comment by alok** [comment]
I made the mistake of actually committing the linting changes

**Comment by alok** [comment]
Check out https://github.com/alok/ray/tree/yapf-take-2 for a clean version

**Comment by richardliaw** [comment]
OK great `take-2` looks a lot better. Consider making a commit that reverts changes from 
6d2a24b onwards and then rebasing `take-2` onto this.

**Comment by alok** [comment]
@richardliaw I ended up making a new branch from master and applying the necessary changes there. See #2083.

**Pull request closed without merging**